static/app/components/autoplayVideo.tsx

```diff
@@ -11,40 +9,29 @@ type Props = React.HTMLProps<HTMLVideoElement>;
  * Note, video needs `muted` for `autoplay` to work on Chrome
  * See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video
  */
-class AutoplayVideo extends React.Component<Props> {
-  componentDidMount() {
-    if (this.videoRef.current) {
+function AutoplayVideo(props: React.VideoHTMLAttributes<HTMLVideoElement>) {
+  const videoRef = React.useRef<HTMLVideoElement>(null);
+
+  React.useEffect(() => {
+    if (videoRef.current) {
       // Set muted as more browsers allow autoplay with muted video.
       // We can't use the muted prop because of a react bug.
       // https://github.com/facebook/react/issues/10389
       // So we need to set the muted property then trigger play.
-      this.videoRef.current.muted = true;
-      const playPromise = this.videoRef.current.play();
+      videoRef.current.muted = true;
 
       // non-chromium Edge and jsdom don't return a promise.
-      playPromise?.catch(() => {
+      videoRef.current.play()?.catch(() => {
         // Do nothing. Interrupting this playback is fine.
       });
     }
-  }
-  private videoRef = React.createRef<HTMLVideoElement>();
-
-  render() {
-    const {className, src, ...props} = this.props;
+  }, []);
 
-    return (
-      <video
-        className={className}
-        ref={this.videoRef}
-        playsInline
-        disablePictureInPicture
-        loop
-        {...props}
-      >
-        <source src={src} type="video/mp4" />
-      </video>
-    );
-  }
+  return (
+    <video ref={videoRef} playsInline disablePictureInPicture loop {...props}>
```

Comment:
since we are using the source element, shouldn't er omit the src prop here? wouldn't it be better if we destructure the props?

Reply:
Ah, you know what. Since we actually only ever use a single `source`, we can just pass `<video src={src}/>`. Thats even simpler and we dont need any destructuring